File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int ans = 0;
        int total = 0, curr = 0;
        for(char ch : s.toCharArray()) total += ch - '0';
        for(int i=0; i<s.length()-1; i++) {
            char ch = s.charAt(i);
            curr += (ch - '0') ^ 1;
            total -= ch - '0';
            ans = Math.max(ans, curr + total);
        }
        return ans;
    }
}

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function(s) {
    let ans = 0, curr = 0, total = 0;
    for(let ch of s) total += ch - '0';
    for(let i=0; i<s.length-1; i++) {
        curr += (s[i] - '0') ^ 1;
        total -= s[i] - '0';
        ans = Math.max(ans, curr + total);
    }
    return ans;
};